org-code-javabuilder/protocol/src/main/java/org/code/protocol/CachedResources.java

```diff
@@ -0,0 +1,49 @@
+package org.code.protocol;
+
+import java.awt.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+// Resources we want to use across multiple invocations of Javabuilder.
+public class CachedResources {
+  private Map<String, Font> fontMap;
+
+  private static final String FONT_FOLDER_NAME = "fonts";
+  private static final String[] FONT_FILE_NAMES =
+      new String[] {
+        "LiberationMono-Bold.ttf",
+        "LiberationMono-BoldItalic.ttf",
+        "LiberationMono-Italic.ttf",
+        "LiberationMono-Regular.ttf",
+        "LiberationSans-Bold.ttf",
+        "LiberationSans-BoldItalic.ttf",
+        "LiberationSans-Italic.ttf",
+        "LiberationSans-Regular.ttf",
+        "LiberationSerif-Bold.ttf",
+        "LiberationSerif-BoldItalic.ttf",
+        "LiberationSerif-Italic.ttf",
+        "LiberationSerif-Regular.ttf"
+      };
+
+  public Map<String, Font> getFontMap() {
```

Comment:
Rather than creating the font map here, could we make it in FontHelper and then set it here? I'm thinking CachedResources could end up with a lot of imports and logic that are related to the item it is caching rather than to the caching itself.

Reply:
we can't have a reference to `media` from `protocol` because `media` references `protocol`, and trying to do the opposite causes a circular reference error. It does end up with some code duplication across the two modules, I tried to keep this as limited in scope as possible. Let me know if you have ideas for a better pattern.

Reply:
we could pass in the map and file names as parameters, but I think the files themselves will need to live in `protocol` since we are importing them here. So it makes more sense to have the filenames live in this class.

Reply:
How about a third class like `FontLoader` that only takes care of loading fonts into memory? We'd still have Font-specific logic in `protocol` but it would at least be isolated to its own class?

Reply:
yeah I can refactor out to a separate class inside protocol.

Reply:
This seems like a bit of a yellow flag.  Is the problem that Font things aren't really part of the `protocol` but we'd like to put it in GlobalProtocol for convenience?  If that's the case, did we consider having CachedResources just be separate (and not part of GlobalProtocol)?  (Side note: this is the kind of thing that Spring dependency injection is supposed to make easy.  Not advocating that we switch to it, but I initially found it hard to understand why we would want to use DI and this feels like one of the target use cases.)

Reply:
The problem is that we need to load the fonts from the `LambdaRequestHandler` as a static variable, then access them from within `media`. Right now our way of sharing resources we create with student code is to put them in the global protocol instance. What do you mean by having CachedResources be separate?

Reply:
ah do you mean we get a static instance of CachedResources separate from our static instance of GlobalProtocol? That might be cleaner